plutus-core/plutus-core/src/PlutusCore/Default/Builtins.hs

```diff
@@ -2111,25 +2112,30 @@ instance uni ~ DefaultUni => ToBuiltinMeaning uni DefaultFun where
             (runCostingFunTwoArguments . paramDropList)
 
     toBuiltinMeaning _semvar LengthOfArray =
-      let lengthOfArrayDenotation :: SomeConstant uni (Vector a) -> BuiltinResult Int
+      let lengthOfArrayDenotation :: SomeConstant uni (ArrayCostedByLength a) -> BuiltinResult Int
```

Comment:
This may look as if the costing function has to measure the length of the array in order to work out the cost of calling the builtin that measures the length of the array, which might seem a bit silly.  However, if the costing function is constant I think the size measure will never actually be evaluated so it's kind of sane after all.

Reply:
Why make it `ArrayCostedByLength` then, why not just `Vector`?